osf/models/node.py

```diff
@@ -2317,6 +2317,7 @@ def on_update(self, first_save, saved_fields):
             # .preprints wouldn't return a single deleted preprint
             for preprint in PreprintService.objects.filter(node_id=self.id, is_published=True):
                 enqueue_task(on_preprint_updated.s(preprint._id, update_share=update_share))
+                preprint.save()
```

Comment:
Driveby comment; IIRC preprint.save() is going to call on_preprint_updated, so this will call on_preprint_updated twice every time a node is modified.